vsomeip-sys/build.rs

```diff
@@ -134,81 +143,72 @@ fn main() -> miette::Result<()> {
-        );
-        line = line.replace(
-            "pub unsafe fn set_data(self: Pin<&mut payload>, _data: *const u8, _length: u32);",
-            "pub(crate) unsafe fn set_data(self: Pin<&mut payload>, _data: *const u8, _length: u32);"
-        );
-        // line = line.replace(
-        //     "# [repr (u8)] # [derive (Clone , Hash , PartialEq , Eq)] pub enum message_type_e",
-        //     "# [repr (u8)] # [derive (Clone , Hash , PartialEq , Eq, Debug)] pub enum message_type_e"
-        // );
+        line = fix_unused_imports(line);
+        line = fix_unsafe_fn_unused(line);
+        line = fix_doc_build(line);
+        line = add_enum_debug(line);
+
         writeln!(writer, "{}", line).expect("Failed to write a line to the temporary file");
     }
 
     writer.flush().expect("Failed to flush the writer buffer");
     fs::rename(temp_file_path, file_path)
         .expect("Failed to rename the temporary file to the original file");
 
-    // let input_file = File::open(&file_path).expect("Unable to find autocxx generated file");
-    // let reader = BufReader::new(input_file);
-    // let mut modified_lines = Vec::new();
-    //
-    // for line in reader.lines() {
-    //     let mut line = line.expect("Unable to read line");
-    //     line = line.replace(
-    //         "pub use bindgen :: root :: std_chrono_duration_int64_t_AutocxxConcrete ;",
-    //         "#[allow(unused_imports)] pub use bindgen :: root :: std_chrono_duration_int64_t_AutocxxConcrete ;"
-    //     );
-    //     line = line.replace(
-    //         "pub use super :: super :: bindgen :: root :: std :: chrono :: seconds ;",
-    //         "#[allow(unused_imports)] pub use super :: super :: bindgen :: root :: std :: chrono :: seconds ;"
-    //     );
-    //     line = line.replace("pub unsafe fn create_payload1", "unsafe fn create_payload1");
-    //     line = line.replace("successfully [de]registered", "successfully de/registered");
-    //     line = line.replace(
-    //         "# [repr (u8)] # [derive (Clone , Hash , PartialEq , Eq)] pub enum message_type_e",
-    //         "# [repr (u8)] # [derive (Clone , Hash , PartialEq , Eq, Debug)] pub enum message_type_e"
-    //     );
-    //     modified_lines.push(line);
-    // }
-    //
-    // let output_file = File::create(&file_path).expect("Unable to write out autocxx modified file");
-    // let mut writer = BufWriter::new(output_file);
-    //
-    // for line in modified_lines {
-    //     writeln!(writer, "{}", line).expect("Unable to write line");
-    // }
-    //
-    // writer.flush().expect("Unable to flush writer for file");
-
     println!("cargo:warning=# rewrote the autocxx file");
 
     Ok(())
 }
 
+fn fix_unused_imports(line: String) -> String {
+    let mut fixed_line = line.replace(
+        "pub use bindgen::root::std_chrono_duration_long_AutocxxConcrete",
+        "#[allow(unused_imports)] pub use bindgen::root::std_chrono_duration_long_AutocxxConcrete",
```

Comment:
Am I right in assuming that the generated code itself does not need the type but that the vsomeip transport will need it?

Reply:
I think we don't use it in the transport. Would you recommend removal entirely then?

Reply:
Unless you want to create a generic vsomeip-sys crate that is independent of vsomeip-utransport ...

Reply:
No thanks, too much work atm :melting_face: 

Will remove

Reply:
Done